### PR TITLE
Ui filters

### PR DESCRIPTION
**Pull request by zrpnr**
Add a checkbox on each attribute for toggling its sparse fieldset.
Removed the hardcoded examples and replaced with the current value of `fields`, along with "clear" buttons.

### CODE REVIEW

File: src/resource.js
Comment (gabesullice) [inline src/resource.js:16]: nice! I considered the same :)

File: src/resource.js
Comment (gabesullice) [inline src/resource.js]: Shouldn't the button text be the field name?

File: src/resource.js
Comment (gabesullice) [inline src/resource.js]: s/Fields/fields

File: src/schema.js
Comment (gabesullice) [inline src/schema.js]: `extract` from the utils file might be more robust since it doesn't create a JS error if the value isn't present. Might not be worth it though.

File: src/schema.js
Comment (gabesullice) [inline src/schema.js]: Nit: could `data` be `fields` instead? I think it's a little confusing to use `data` since that key is a big part of the spec and means something different.

File: src/attribute.js
Comment (gabesullice) [inline src/attribute.js]: Nice new util.

File: src/resource.js
Comment (gabesullice) [reply src/resource.js]: I like that this also used the type name and that the sentence is more human friendly by saying _which_ fields are being cleared.

**Review by gabesullice — APPROVED** [review]
Nice! This is a big step forward for the UI. I had a few small things. We can fix those after merge or before (if @zrpnr agrees w/ them). I'll let you decide @zrpnr.

File: src/resource.js
Comment (zrpnr) [reply src/resource.js]: should the "clear" verb be there still? Since although it is toggling it is only ever toggling "off" from here.

File: src/schema.js
Comment (zrpnr) [reply src/schema.js]: I'm fine with not calling it `data` but in the attribute and relationship components I'm using the "context" `fields` so there's a conflict. Do you have another suggestion?

Comment (zrpnr) [reply src/schema.js]: maybe just `attributes={attributes}` ?

File: src/resource.js
Comment (zrpnr) [reply src/resource.js]: Added the field name into the button

File: src/resource.js
Comment (zrpnr) [reply src/resource.js]: Fixed capitalization

File: src/schema.js
Comment (zrpnr) [reply src/schema.js]: Good idea, always better to not assume a property exists.

**Comment by zrpnr** [comment]
Thanks for the review, glad you approved this update.

**Pull request merged**
